### ISSUE TITLE
Shimming Twitter Bootstrap breaks with Browserify > 4.1.9

### ISSUE THREAD

**Issue by Pisi-Deff** [thlorenz/browserify-shim#138]
Greetings.

I am on Win7 x64
And with Node v0.12.0

I spent a lot of time the past few weeks trying to figure out why browserify-shim seemed to be ignoring its configuration, and not adding a require('jquery') for bootstrap.

On my path, I found this repo: https://github.com/Chandu/browserify-test
To my surprise, that worked perfectly, even though it was very close to my setup.

I started trying out different versions of browserify and browserify-shim.

I ended up noticing that anything that I tried up until browserify 4.1.9 works, anything I tried after that fails. The browserify-shim version remained constant, at 3.8.2.

### Output with browserify > 4.1.9 (including 8.1.3)
    { file: '[...]\\browserify-test\\assets\\common.js',
      info:
       { package_json: '[...]\\browserify-test\\package.json',
         packageDir: '[...]\\browserify-test',
         shim: undefined,
         exposeGlobals: {},
         browser: { bootstrap: './node_modules/bootstrap/dist/js/bootstrap.js' },
         'browserify-shim': { bootstrap: { exports: 'bootstrap', depends: [ 'jquery:jQuery' ] } },
         dependencies: { jquery: '~2.1.1', bootstrap: '~3.2.0' } },
      messages:
       [ 'Found "bootstrap" in browser field referencing "./node_modules/bootstrap/dist/js/bootstrap.js" and resolved it to
    "[...]\\browserify-test\\node_modules\\bootstrap\\dist\\js\\bootstrap.js"',
         'Found depends "jquery" as an installed dependency of the package',
         { resolved: { '[...]\browserify-test\node_modules\bootstrap\dist\js \bootstrap.js': { exports:
     'bootstrap', depends: { jquery: 'jQuery' } } } } ] }
This does not result in a jquery require for bootstrap.

### Output with browserify = 4.1.9
    { file: '[...]\\browserify-test\\assets\\common.js',
      info:
       { package_json: '[...]\\browserify-test\\package.json',
         packageDir: '[...]\\browserify-test',
         shim: undefined,
         exposeGlobals: {},
         browser: { bootstrap: './node_modules/bootstrap/dist/js/bootstrap.js' },
         'browserify-shim': { bootstrap: { exports: 'bootstrap', depends: [ 'jquery:jQuery' ] } },
         dependencies: { jquery: '~2.1.1', bootstrap: '~3.2.0' } },
      messages:
       [ 'Found "bootstrap" in browser field referencing "./node_modules/bootstrap/dist/js/bootstrap.js" and resolved it to
    "[...]\\browserify-test\\node_modules\\bootstrap\\dist\\js\\bootstrap.js"',
         'Found depends "jquery" as an installed dependency of the package',
         { resolved: { '[...]\browserify-test\node_modules\bootstrap\dist\js\bootstrap.js': { exports:
     'bootstrap', depends: { jquery: 'jQuery' } } } } ] }

    { file: '[...]\\browserify-test\\node_modules\\bootstrap\\dist\\js\\bootstrap.js',
      info:
       { package_json: '[...]\\browserify-test\\package.json',
         packageDir: '[...]\\browserify-test',
         resolvedPreviously: true,
         shim: { exports: 'bootstrap', depends: { jquery: 'jQuery' } },
         exposeGlobals: {},
         browser: { bootstrap: './node_modules/bootstrap/dist/js/bootstrap.js' },
         'browserify-shim': { bootstrap: { exports: 'bootstrap', depends: [ 'jquery:jQuery' ] } },
         dependencies: { jquery: '~2.1.1', bootstrap: '~3.2.0' } },
      messages: [] }
This results in:
```; jQuery = global.jQuery = require("jquery");
    ;__browserify_shim_require__=require;(function browserifyShim(module, exports, require, define, browserify_shim__define__module__export__) {```

Now, what's more confusing is that if I just use 
`browserify ./assets/common.js -o public/common.js`
instead of `gulp`, I get the first result - jquery dep not added for bootstrap. This is for all 4.1.9, 4.1.10, 8.1.3 of browserify.

I have no clue whether this is a browserify-shim or browserify issue but any help would be appreciated.

**Comment by bendrucker** [comment]
Please upload a repo we can clone and run to reproduce this with a single command (other than `npm install`).

**Comment by Pisi-Deff** [comment]
Before I spend time on that, just to make sure, would just using the abovementioned repo work for you?
https://github.com/Chandu/browserify-test

`npm install`
`npm install --save-dev browserify@4.1.9`
`gulp`
./public/common.js line # 16 is: `; jQuery = global.jQuery = require("jquery");`

`npm install --save-dev browserify@4.1.10` or 8.1.3
`gulp`
./public/common.js line # 16 is part of bootstrap copyright comment which previously was below the jquery require

**Comment by bendrucker** [comment]
> To my surprise, that worked perfectly, even though it was very close to my setup.

Yes. It's really difficult to parse that huge block of text and I really don't want to go making changes to the repo you've linked.

**Comment by Pisi-Deff** [comment]
Minimal (I think) example for you:
https://github.com/Pisi-Deff/browserify-shim-test
Browserify@4.1.9 is initially marked as devDep in package.json

`npm install`
`gulp`
./bundle.js line # 15 is: `; jQuery = global.jQuery = require("jquery");`

`npm install --save-dev browserify@4.1.10` or 8.1.3
`gulp`
./bundle.js line # 15 is part of bootstrap copyright comment which previously was below the jquery require and jquery require is _missing_.

**Comment by bendrucker** [comment]
Please just focus on what's not working right now. Doesn't really matter to much that v4 used to work.

**Comment by Pisi-Deff** [comment]
I'm sorry, I do not understand what you want me to do.

**Comment by bendrucker** [comment]
Your reproduction looks fine for now so no need to do anything. Don't worry about this working previously. That's gonna end up being too hard to track down.

**Comment by Pisi-Deff** [comment]
Ah, excellent.
I hope you can come up with something.
Thank you!

**Comment by bendrucker** [comment]
Just tried this and had no issues with b@8. Logs that function without any issues when executed.

**Comment by Pisi-Deff** [comment]
Yet definitely still does not work for me.
Tried at home as well, which didn't have node.js installed until today.

**Comment by bendrucker** [comment]
Yeah no idea what the issue would be there. Build your bundle and upload it as a gist so I can compare against the OS X/Unix output.

**Comment by Pisi-Deff** [comment]
Here you go:
https://gist.github.com/Pisi-Deff/9ff1d0ab25d824cc1b56

**Comment by bendrucker** [comment]
Sorry, no idea why shimming isn't working properly in Windows. I don't have access to a Windows machine to look into this.

**Comment by anatoliyarkhipov** [comment]
@bendrucker @Pisi-Deff your duscussion guided me to try to build it on the Ubuntu virtual machine, and after I did it once, build began to work properly on Windows 7. :/

Now I've changed some scripts in the shim and it still working good, I can't break it back.

**Comment by anatoliyarkhipov** [comment]
Ok, I got it. First time I missed one moment. When I tried to build it on Ubuntu it throws error `Can't find module, bla-bla`. It was because I forgot the `.js` extension:

```
"browser": {
    "bootstrap": "./resources/js/bower_components/bootstrap/dist/js/bootstrap",
```

But when I run the build process on Windows, it does not throw any errors about incorrect path and just builds that, but without properly shimming.

So, when I did this first time on Ubuntu, I fixed the extension, and from this moment it works fine on Windows.
